### PR TITLE
Splash component and variables for buttons

### PR DESCRIPTION
**Pull request by slafleche**
This PR is for the "Splash" component, featured on the KB's home page.

There were a few rabbit holes along the way. I did a lot of work to move over the button styles I needed, as well as a new generic "search component" and the backLink styles.

### CODE REVIEW

**Comment by slafleche** [comment]
Found a few bugs. I'm fixing them now.

File: library/src/scripts/components/Search.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    cancelContentClassName?: string;
+    theme?: object;
+}
+
+interface IState {
+    query: string;
+}
+
+/**
+ * Implements Compact Search component for header
```
Comment (charrondev) [inline library/src/scripts/components/Search.tsx]: Could we update this docblock?

Comment (charrondev) [reply library/src/scripts/components/Search.tsx]: I'm also thinking this class name could be more specific. Maybe `IndependentSearch` or `SplashSearch` and moved to the splash directory.

**Comment by charrondev** [comment]
1 other thing I'm noticing is that the `cursor` value seems to be `pointer` over the entire search input. It should be `text` while hovering over the text input and `pointer` while hovering over the button or results.

**Pull request merged**